plugins/libplugin.c

```diff
@@ -1448,12 +1448,20 @@ static struct plugin *new_plugin(const tal_t *ctx,
 	p->in_timer = 0;
 
 	p->commands = commands;
+	if (taken(commands))
+		tal_steal(p, commands);
```

Comment:
I'm surprised that this works with non-tal allocated arrays too. I would have guessed that `taken()` on `commands` from `keysend.c` for example, which is statically allocated, would fail when looking up the tal header.

Reply:
?  The only one which passes take(commands) is spender().

But yes, taken() works on *any* pointer: take(x) actually puts x into a hash table...

Reply:
I see, I was under the impression that `take` was using the `tal_header`. Thanks for the clarification.